### PR TITLE
Add 'jupyter kernel' command

### PR DESCRIPTION
**Pull request by takluyver**
A simple lead in to the 'kernel nanny' work, this adds a command so you can do:

```
jupyter kernel --kernel python
```

I intend this for post-5.0.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Could this be tested?

File: jupyter_client/kernelapp.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+from jupyter_core.application import JupyterApp
+from tornado.ioloop import IOLoop
+from traitlets import Unicode
+
+from . import __version__
+from .kernelspec import KernelSpecManager
+from .manager import KernelManager
+
+class KernelApp(JupyterApp):
```
Comment (minrk) [inline jupyter_client/kernelapp.py]: docstrings anywhere?

**Comment by minrk** [comment]
Very nice! I'd love a test to get basic exercise of the signal stuff.

**Comment by rolweber** [comment]
Is this related to issue #32?

**Comment by takluyver** [comment]
Yep, it is. I've basically stopped trying to work on everything related to the 'kernel nanny' proposal (https://github.com/jupyter/enhancement-proposals/pull/14) for now, though, because I don't have a good enough feel for the problem.

This will probably also need to be updated in light of #261, if that goes forward.

**Comment by takluyver** [comment]
I've come back to this following a recent email. I've added some docstrings and a rather crude test.

**Comment by minrk** [comment]
Great! Thanks for coming back to this one.

**Comment by takluyver** [comment]
@meeseeksdev backport

**Comment by takluyver** [comment]
Backported manually.

**Comment by jakirkham** [comment]
Have you explored using `pytest-runner`? Also have found this [suggestion]( https://github.com/pytest-dev/pytest-runner#conditional-requirement ) helpful when working with `pytest-runner`.

**Pull request merged**